src/middleware/persist.ts

```diff
@@ -312,7 +312,7 @@ export const persist =
           }
         })
         .then((migratedState) => {
-          stateFromStorage = options.merge(migratedState as S, configResult)
+          stateFromStorage = options.merge(migratedState as S, get())
```

Comment:
Maybe this could fix your issue and the tests?

```suggestion
          stateFromStorage = options.merge(migratedState as S, configResult.getState())
```

Reply:
`configResult` is the initialState object and it doesn't have a `getState()` method.

Reply:
Fixed tests and added my own test case.